src/main/java/cn/nukkit/block/BlockSponge.java

```diff
@@ -112,13 +112,13 @@ private boolean performWaterAbsorb(Block block) {
                     if (faceBlock.getId() == BlockID.BLOCK_KELP || faceBlock.getId() == BlockID.SEAGRASS || faceBlock.getId() == BlockID.SEA_PICKLE || faceBlock instanceof BlockCoralFan) {
                         faceBlock.getLevel().useBreakOn(faceBlock);
                     }
-                    this.getLevel().setBlockStateAt(faceBlock.getFloorX(), faceBlock.getFloorY(), faceBlock.getFloorZ(), 1, Block.get(BlockID.AIR));
+                    this.getLevel().setBlockStateAt(faceBlock.getX(), faceBlock.getY(), faceBlock.getZ(), 1, Block.get(BlockID.AIR));
```

Comment:
The whole basis of this pull request relies on the change from `floorX` to `X` affecting the sponge's behaviour.

Why is that the case and why floored coordinates would be a problem?

As @LoboMetalurgico stated there is no reason why the simple change to exact coordinates would change any behaviour.

For this reason, until this change is explained or we're able to notice the fixes at all we can't merge this.